src/gc/env/gcenv.ee.h

```diff
@@ -46,14 +46,14 @@ class GCToEEInterface
     static void SyncBlockCachePromotionsGranted(int max_gen);
 
     // Thread functions
-    static bool IsPreemptiveGCDisabled(Thread * pThread);
-    static void EnablePreemptiveGC(Thread * pThread);
-    static void DisablePreemptiveGC(Thread * pThread);
+    static bool IsPreemptiveGCDisabled();
+    static void EnablePreemptiveGC();
+    static void DisablePreemptiveGC();
     static bool TrapReturningThreads();
     static Thread* GetThread();
```

Comment:
Do we really need GetThread for anything?

Reply:
We still use it for flow control in a few places, e.g.

* Asserting whether or not a function is called on a bgc thread
* Asserting whether or not we can take the finalizer queue lock (probably can be refactored... I'll try in next iteration)
* `WaitLongerNoInstr` has different control flow depending on whether not `GetThread` is null - I wasn't familiar with why this was done so I left it be.
* Keeping track of which thread holds spin locks in debug builds
* One use in `gc_heap::gc1` that I think can be refactored

I'll see what I can do with these but it seems to me that all non-assert uses of `GetThread` are probably not necessary.